### ISSUE TITLE
Phase11-Part1-Amirhossein-Arabzadeh

### ISSUE THREAD

**Issue by amirarab888** [Star-Academy/codestar-intern-issues#182]
This issue template is based of [Phase11-Web.md](./Phase11-Web.md) contents.
Click on "Raw" button on right side of this page to see original markdown then create a new issue with original content of this tasklist:

# Phase 11 First Part

- [x] Before starting: Read the link about history and purpose of creating & developing Web, then answer this questions:
  - [ ] How did developers provide server-client softwares for users before creation of web technologies?
  
      `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
      
  - [ ] What was the main trouble with traditional technologies before web?
  
      `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
      
  
  - [ ] How did web overcome this main trouble?
  
      `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`

## Asp.net core

- [ ] Getting Started with HTTP:
  - [ ] What does a protocol mean?
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`

  - [ ] Why designing a protocol for communication is important prerequirement of all other web related technologies?
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
    
  - [ ] Try to get familier with HTTP by reading links.
  - [ ] Why HTTP is most popular protocol in web technologies?
    
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
    
  - [ ] What is the most interesting aspect of HTTP in your opinion? 
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`

- [ ] Getting Started with REST:
  - [ ] Read mentioned links.
  - [ ] Why do we need RESTful APIs?
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`

- [ ] Getting Started with Asp.net Core:
  - [ ] Read mentioned links.
  - [ ] Create a hello world API.
  - [ ] Call your api from browser.
  - [ ] Play with routing options! (Try to use it & enjoy it!)
  - [ ] Why do we need configurable routings in web development?
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
  
  - [ ] Play with dependency injection! (Define an interface & inject interface implementation to code)
  - [ ] Why do we need dependency injection?
  
    `[FILL HERE WITH YOUR ANSWER. (feel free to write in persian)]`
  - [ ] Play with parameter binding! (use different modes)
  - [ ] Install postman.
  - [ ] Call your APIs with postman.
  
- [ ] Implement Asp.net Core Project:
  - [ ] Design your search API.
  - [ ] Implement & Test your API.

# Review
Link of your own PR:

`[FILL HERE WITH LINK OF YOUR PR.]`

Link of other PR(s) you agreed to review:

`[FILL HERE WITH LINK OF PR(s) YOU REVIEWED.]`

- [ ] Your own PR is reviewed and approved by at least one other team.

- [ ] Your own PR is reviewed and approved by mentor (hamyar).

- [ ] Your own PR is reviewed and approved by manager (rahbar).

- [ ] Your own PR is merged.
